sklearn/_build_utils/__init__.py

```diff
@@ -75,3 +75,26 @@ def maybe_cythonize_extensions(top_path, config):
             nthreads=n_jobs,
             compile_time_env={'SKLEARN_OPENMP_SUPPORTED': with_openmp},
             compiler_directives={'language_level': 3})
+
+
+def gen_from_templates(templates):
+    """Generate cython files from templates"""
+    if not isinstance(templates, list):
+        templates = [templates]
+
+    for template in templates:
+        outfile = template.replace('.tp', '')
+
+        if not (os.path.exists(outfile) and
+                os.stat(template).st_mtime < os.stat(outfile).st_mtime):
```

Comment:
Comment got marked resolved because of update:

I think we should remove this, since this suggests that we can directly edit the `pyx` files which is wrong